### PR TITLE
[STFT] Pytorch FE enablement of STFT operation

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 -  Pytorch FE enablement of STFT op
 - This PR contains STFT CPU enablement changes (to be merged first), please review pytorch related changes only within this PR:
 src/frontends/pytorch/src/op/stft.cpp + test_stft.py

### Tickets:
 - 155166, 143017
 
 Related PR:
 - https://github.com/openvinotoolkit/openvino/pull/27137

### CODE REVIEW

File: src/frontends/pytorch/src/op/stft.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        onesided = context.const_input<bool>(6);
+    }
+    PYTORCH_OP_CONVERSION_CHECK(onesided, "aten::stft conversion is currently supported with onesided=True only.");
+
+    bool return_complex = false;
+    if (!context.input_is_none(7)) {
+        return_complex = context.const_input<bool>(7);
+    }
+    PYTORCH_OP_CONVERSION_CHECK(!return_complex,
+                                "aten::stft conversion is currently supported with return_complex=False only.");
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/stft.cpp:82]: I see these 3 attributes are not supported, do we have plans to support them in future?

Comment (eaidova) [reply src/frontends/pytorch/src/op/stft.cpp:82]: that is probably required support complex numbers for pytorch frontend?

File: tests/layer_tests/pytorch_tests/test_stft.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from pytorch_layer_test_class import PytorchLayerTest
+
+
+class TestSTFT(PytorchLayerTest):
+    def _prepare_input(self, win_length, signal_shape, rand_data=False, out=False, out_dtype="float32"):
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_stft.py:10]:
```suggestion
    def _prepare_input(self, win_length, signal_shape, rand_data=False, out_dtype="float32"):
```
`out` is unused

File: src/frontends/pytorch/src/op/stft.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    auto input = context.get_input(0);
+    auto n_fft = context.get_input(1);
+
+    ov::Output<ov::Node> hop_length;
+    if (!context.input_is_none(2)) {
+        hop_length = context.get_input(2);
+    } else {
+        // Defualt floor(n_fft / 4)
+        const auto four = context.mark_node(std::make_shared<ov::op::v0::Constant>(ov::element::i32, Shape{}, 4));
```
Comment (eaidova) [inline src/frontends/pytorch/src/op/stft.cpp:39]: include header for constant missed

**Review by mvafin — APPROVED** [review]
Looks good from PT FE side.
Note: STFT torch implementation can return complex numbers while our implementation return only real part of the number, this will require spec change in future to support this case.

File: src/frontends/pytorch/src/op/stft.cpp (hunk 1 above)
Comment (mvafin) [reply src/frontends/pytorch/src/op/stft.cpp:82]: Not really, complex number can be represented as a tensor with last dim as 2

**Review by eaidova — APPROVED** [review]

File: src/frontends/pytorch/src/op/stft.cpp (hunk 1 above)
Comment (mitruska) [reply src/frontends/pytorch/src/op/stft.cpp:82]: But it is, the last dimension of the STFT output is indeed 2, to represent real and imaginary part.

**Comment by mitruska** [comment]
> Looks good from PT FE side. Note: STFT torch implementation can return complex numbers while our implementation return only real part of the number, this will require spec change in future to support this case.

It return real and imaginary part, but separately as a two float numbers (not a single complex type).

**Comment by mvafin** [comment]
> > Looks good from PT FE side. Note: STFT torch implementation can return complex numbers while our implementation return only real part of the number, this will require spec change in future to support this case.
> 
> It returns real and imaginary part, but separately as a two float numbers (not a single complex type).

Agree, disregard my note then

**Review by mmikolajcz — APPROVED** [review]

File: tests/layer_tests/pytorch_tests/test_stft.py (hunk 2 above)
Comment (mitruska) [reply tests/layer_tests/pytorch_tests/test_stft.py]: Removed

File: src/frontends/pytorch/src/op/stft.cpp (hunk 3 above)
Comment (mitruska) [reply src/frontends/pytorch/src/op/stft.cpp:40]: Added

**Pull request merged**